templates/todo/projects/csharp-cosmos-sql/.repo/bicep/infra/main.bicep

```diff
@@ -141,6 +145,40 @@ module monitoring '../../../../../../common/infra/bicep/core/monitor/monitoring.
     logAnalyticsName: !empty(logAnalyticsName) ? logAnalyticsName : '${abbrs.operationalInsightsWorkspaces}${resourceToken}'
     applicationInsightsName: !empty(applicationInsightsName) ? applicationInsightsName : '${abbrs.insightsComponents}${resourceToken}'
     applicationInsightsDashboardName: !empty(applicationInsightsDashboardName) ? applicationInsightsDashboardName : '${abbrs.portalDashboards}${resourceToken}'
+    useAPIM: useAPIM
+  }
+}
+
+resource appInsightsService 'Microsoft.ApiManagement/service@2021-08-01' existing = {
```

Comment:
Do we need to expose this resource here in `main`? Feels like we should opt to do what the standard `api` app serivce is using where we accept the app insights name as a param and have any `existing` resource referenced internally within that module?

I believe this builds on an early comment about not exposing the app insights resource id as a param and just do the lookup internally if possible.

Reply:
I understand. I will fix it

Reply:
fixed and tested with success